spinta/manifests/tabular/helpers.py

```diff
@@ -1524,6 +1530,68 @@ def leave(self) -> None:
 }
 
 
+class NestedPropertyNode:
+    name: str
+    full_name: str
+    full_parent_name: Optional[str]
+    depth: int = 0
+    data: PropertyRow
+
+    def __init__(self, given_name: str, data: PropertyRow):
+        self.full_name = given_name
+        self.data = data
+        self.name, self.full_parent_name = self._split_name(given_name)
+        self.depth = self._depth(given_name)
+
+    def is_parent_of(self, other_node: NestedPropertyNode) -> bool:
+        if (name := other_node.full_parent_name) and name == self.full_name:
+            return True
+        return False
+
+    def _split_name(self, given_name: str) -> tuple[str, Optional[str]]:
+        parts = given_name.split('.')
+        full_parent_name = '.'.join(parts[:-1]) if parts[:-1] else None
+        return parts[-1], full_parent_name
+
+    def _depth(self, given_name: str) -> int:
+        return given_name.count('.')
+
+
+class InvalidPropertyHierarchyError(Exception):
```

Comment:
Exceptions should be added here https://github.com/atviriduomenys/spinta/blob/master/spinta/exceptions.py  and extend BaseError, UserError or some other base exception class. For more details probably ask Mantas :)